### DISCUSSION TITLE
Pass through environmental variables to files called

### DISCUSSION THREAD

**Discussion by transcental** [community/community#9640] [Actions] [answered]
When a GitHub action runs, it would be good if the environmental variables that exist are able to be used in the file that is called. For example I might have a `workflow.py` file that I want running every so often so I automate it via GH Actions.

These could be the contents:

```py
import os

token = os.environ['TOKEN']

function(token)
```

This should be able to work when running via a workflow.

**Comment by chrispat** [comment]
All [environment variables that are defined](https://docs.github.com/en/actions/learn-github-actions/environment-variables) are passed to when running something from a `run` step and as such you should be able to access them.

**Reply by transcental** [reply]
Sorry for the late response.

I've looked at the documentation you linked, but it isn't very clear on what you do. I've got a repository secret called `PERSONAL_ACCESS_TOKEN`. As I understand it, this is in the `$GITHUB_ENV` file, but I don't understand how you make the secret available for a specific job.

**Reply by transcental** [reply]
To be a bit clearer, I get that you set envs like so:

```yml
env:
  NAME: value
```
 
But I obviously don't want to put my token in the file, so how do I read it from wherever GitHub stored it?

> EDIT: [Solved](https://github.com/github/feedback/discussions/9640#discussioncomment-1957675) but the documentation really isn't very clear on this.

**Comment by transcental** [comment, marked as answer]
Solved thanks to this post: https://bloggie.io/@_junrong/using-environment-variables-secrets-in-github-actions

For reference, you can do it like this: 
```yml
env:
  PERSONAL_ACCESS_TOKEN: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
```
